ostree/ostree_src.go

```diff
@@ -348,7 +364,45 @@ func (s *ostreeImageSource) GetSignatures(ctx context.Context, instanceDigest *d
 	return signatures, nil
 }
 
-// LayerInfosForCopy() returns updated layer info that should be used when reading, in preference to values in the manifest, if specified.
+// LayerInfosForCopy() returns the list of layer blobs that make up the root filesystem of
+// the image, after they've been decompressed.
 func (s *ostreeImageSource) LayerInfosForCopy() []types.BlobInfo {
-	return nil
+	updatedBlobInfos := []types.BlobInfo{}
+	manifestBlob, manifestType, err := s.GetManifest(nil)
+	if err != nil {
+		return nil
```

Comment:
Not sure about some other cases, but at least in this one it seems clear that `err` shouldn’t be thrown away; can you add an `error` return value to `LayerInfosForCopy`?

Reply:
this is the signature from the `ImageSource interface.  The only user of this function in `ostree_src` is `GetBlob` and I handle there the `nil` return, which is considered an error.

Reply:
Right; please change the interface. `c/i/copy/imageCopier.copyLayers` should be able to tell the difference between “no changes needed” and “reading manifest failed” / ”manifest invalid”.

Reply:
I've pushed a new change to change the interface